### PR TITLE
create application mutation

### LINKED ISSUE #3332: Backend mutations for Creating Applications
# Description

## Create Draft mutation
Create a mutation which creates a stub application, aka a PoolCandidate, for the currently logged-in user and the specified pool. This PoolCandidate object can be devoid of any data besides the pool, the user, and something that marks it as a Draft (a new status?). 

QUESTION: if the user already has a PoolCandidate for the pool, do we want to throw an error or just return the pool that already exists?

ANSWER: (Gray) I think so... it seems like returning the existing one will solve more problems then it causes. So yes, unless there's a good reason not to.

# Acceptance Criteria
- [ ] Create mutation creates a new PoolCandidate object with very little data, but in a draft status
- [ ] Draft status simply means that the 'submitted_at' field is null (potentially)
- [ ] If user already has a PoolCandidate for the pool, create mutation returns that existing object instead of creating a new one.

### PR DESCRIPTION
**Pull request by vd1992**
resolves #3332 

testing: 
use this mutation
you will need to fill in values from adminer, if you attempt to create a duplicate application the row count should remain unchanged, and should increment by one when you attempt to make a new one
```
mutation createApplication {
  createApplication(userId: "USER_ID", poolId: "POOL_ID") {
    user {
      id
    }
    pool {
      id
    }
  }
}
```

status work I did not complete as #3660 deals with that, adding DRAFT as well as doing the status logic
so what I did was just implement the submitted_at field for that issue

### CODE REVIEW

File: api/app/GraphQL/Mutations/CreateApplication.php
Comment (esizer) [inline api/app/GraphQL/Mutations/CreateApplication.php:24]: If you set these as fillable, you could convert this to:

```php
$application = PoolCandidate::firstOrCreate([
  'user_id' => $args['userId'],
  'pool_id' => $args['poolId']
]);
```

File: api/database/migrations/2022_08_11_161706_submitted_at_field_added_application.php
Comment (esizer) [inline api/database/migrations/2022_08_11_161706_submitted_at_field_added_application.php:17]: Should this have a default of `null` so we don't accidentally create a pool candidate with this set? Or is it implied it is `null`? 

This is more of a question than a suggestion because I'm not sure 🤷

**Review by esizer — APPROVED** [review]
Nice work! 👍 Just a small suggestion and a question from me so I can learn a bit more :)

File: api/database/migrations/2022_08_11_161706_submitted_at_field_added_application.php
Comment (vd1992) [reply api/database/migrations/2022_08_11_161706_submitted_at_field_added_application.php:17]: It shouldn't be accidentally set anywhere, just empty/null I think
hm, let us be cautious 
@petertgiles please enlighten us

Comment (petertgiles) [reply api/database/migrations/2022_08_11_161706_submitted_at_field_added_application.php:17]: If it's nullable and there's no default, then yes it will be null for new rows.  That said, I'm all for making implied things more explicit.

Comment (vd1992) [reply api/database/migrations/2022_08_11_161706_submitted_at_field_added_application.php]: and done
https://github.com/GCTC-NTGC/gc-digital-talent/pull/3662/commits/f86eef5e4d07ed28fc731a03b0731943aeea7278

**Comment by vd1992** [comment]
@esizer 
I made some changes, so I think it would be best if you please quickly glanced over it before I merge

**Review by esizer — APPROVED** [review]
This has my stamp of approval 🎖️

**Pull request merged**
